src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/EnvelopeEventProcessor.java

```diff
@@ -51,6 +51,16 @@ public EnvelopeEventProcessor(
         this.appInsights = appInsights;
     }
 
+    /**
+     * Reads and processes next message from the queue.
+     *
+     * @return false if there was no message to process. Otherwise true.
+     */
+    public boolean processNextMessage() throws ServiceBusException, InterruptedException {
+        // TODO: implement
+        return false;
```

Comment:
Implementation has been extracted (https://github.com/hmcts/bulk-scan-orchestrator/compare/feature/introduce-circuit-breaker), as this is a big enough change alone.